contracts/contracts/system_contracts/kaiabridge/EnumerableSetUpgradable.sol

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.24;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+contract EnumerableSetUpgradable is Initializable {
```

Comment:
Why do you use `contract` for `EnumerableSetUpgrdable`? You can directly use `openzeppelin-contracts-5.0/EnumerableSet.sol` since it's a library. I think implementing custom methods `getAll, getRange` in a wrapper library (if needed by many contracts) will be better.

Reply:
I wanted to wrap the contract as upgradable version. That's why I created a new dedicated contract to achieve upgradable.

Reply:
Sorry for late reply.

When upgrading a contract, it's safe to replace library with additional function since pure library doesn't affect storage layout of contract at all. But if you expect to add some variables to library, then I think you need to add gap since it's inherited by other abstract contracts.

Reply:
Sorry for late reply.

When upgrading a contract, it's safe to replace library with additional function since pure library doesn't affect storage layout of contract at all. But if you expect to add some variables to library, then I think it needs a gap since it's inherited by other abstract contracts.